twilight-gateway/src/shard.rs

```diff
@@ -608,10 +588,18 @@ impl Shard {
     /// let intents = Intents::GUILD_VOICE_STATES;
     /// let token = env::var("DISCORD_TOKEN")?;
     ///
-    /// let mut shard = Shard::new(ShardId::ONE, token, intents).await?;
+    /// let mut shard = Shard::new(ShardId::ONE, token, intents);
     ///
-    /// // Query members whose names start with "tw" and limit the results to
-    /// // 10 members.
+    /// // Discord only allows sending the `RequestGuildMembers` command after
+    /// // the shard is identified.
+    /// while !shard.status().is_connected() {
```

Comment:
This example is why I felt the need to also expand the `ConnectionStatus` variants and semantics in this PR. Without that change there's no way to know when the shard is connected